core/whistle-1.0.0/src/wh_util.erl

```diff
@@ -408,13 +411,96 @@ is_account_expired(Account) ->
         {'error', _R} ->
             lager:debug("failed to check if expired token auth, ~p", [_R]),
             'false';
-        {'ok', Doc} ->
-            Now = ?MODULE:current_tstamp(),
-            Trial = kz_account:trial_expiration(Doc, Now+1),
-            case Trial < Now of
-                'false' -> 'false';
-                'true' -> {'true', Trial}
-            end
+        {'ok', JObj} ->
+             kz_account:is_expired(JObj)
+    end.
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec maybe_disable_account(ne_binary()) ->'ok' | 'failed'.
+maybe_disable_account(Account) ->
+    case is_account_enabled(Account) of
+        'false' -> 'ok';
+        'true' ->
+            disable_account(Account)
+    end.
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec disable_account(ne_binary()) ->'ok' | 'failed'.
+disable_account(Account) ->
+    account_update(Account, fun kz_account:disable/1).
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec enable_account(ne_binary()) ->'ok' | 'failed'.
+enable_account(Account) ->
+    account_update(Account, fun kz_account:enable/1).
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec set_superduper_admin(ne_binary(), boolean()) ->'ok' | 'failed'.
+set_superduper_admin(Account, IsAdmin) ->
+    account_update(Account, fun(J) -> kz_account:set_superduper_admin(J, IsAdmin) end).
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec set_allow_number_additions(ne_binary(), boolean()) ->'ok' | 'failed'.
+set_allow_number_additions(Account, IsAllowed) ->
+    account_update(Account, fun(J) -> kz_account:set_allow_number_additions(J, IsAllowed) end).
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec account_update(ne_binary(), function()) -> 'ok' | 'failed'.
+account_update(Account, Fun) ->
+    Updaters = [
+        fun({'error', _}=E) -> E;
+            ({'ok', J}) ->
+                {'ok', Fun(J)}
+         end
+        ,fun({'error', _}=E) -> E;
+            ({'ok', J}) ->
+                AccountDb = wh_util:format_account_id(Account, 'encoded'),
+                couch_mgr:save_doc(AccountDb, J)
+        end
+        ,fun({'error', _}=E) -> E;
+            ({'ok', J}) ->
```

Comment:
`J` here should have the new revision; no need to do the `lookup_doc_rev` lower down